### PR TITLE
poke: 3.2 -> 4.0

### PR DESCRIPTION
**Pull request by kira-bruneau**
## Description of changes

https://www.jemarch.net/poke-4.0-relnotes.html

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by kira-bruneau** [comment]
aarch64-darwin has been broken since [the update from 2.4 -> 3.2](https://github.com/NixOS/nixpkgs/commit/8bbe4ca114594d7902366ee6f3932151596744da), so for now I just marked it as broken.

It seems like we'll need to add back `autoreconfHook` which was originally required to fix the aarch64-darwin build: https://github.com/mmlb/home-manager-nixpkgs/commit/952c5269b39c31b47a5470c5db9b7fe6a1089675, but I can't seem to get it to work.

File: pkgs/applications/editors/poke/default.nix [hunk 1]
```diff
@@ -97,18 +75,17 @@ in stdenv.mkDerivation rec {
       # Expect the text in format of '<a href="...">poke 2.0</a>'
       new_version="$(curl -s https://www.jemarch.net/poke |
           pcregrep -o1 '>poke ([0-9.]+)</a>')"
-      update-source-version ${pname} "$new_version"
+      update-source-version ${finalAttrs.pname} "$new_version"
```
Comment (Aleksanaa) [inline pkgs/applications/editors/poke/default.nix:78]:
```suggestion
      update-source-version poke "$new_version"
```

File: pkgs/applications/editors/poke/default.nix [hunk 2]
```diff
@@ -40,53 +38,33 @@ in stdenv.mkDerivation rec {
...
-    "--with-tk=${tk}/lib"
-    "--with-tkinclude=${tk.dev}/include"
   ];
 
   enableParallelBuilding = true;
 
-  doCheck = !isCross;
-  nativeCheckInputs = lib.optionals (!isCross) [ dejagnu ];
+  doCheck = true;
+  nativeCheckInputs = lib.optionals finalAttrs.finalPackage.doCheck [ dejagnu ];
```
Comment (Aleksanaa) [inline pkgs/applications/editors/poke/default.nix:61]:
```suggestion
  nativeCheckInputs = [ dejagnu ];
```

If we don't do check, then no `nativeCheckInputs`

Comment (kira-bruneau) [reply pkgs/applications/editors/poke/default.nix:61]: Oh yep thanks! Was wanting to simplify the redundant `isCross` check and forgot this!

File: pkgs/applications/editors/poke/default.nix [hunk 3]
```diff
@@ -97,18 +75,17 @@ in stdenv.mkDerivation rec {
       # Expect the text in format of '<a href="...">poke 2.0</a>'
       new_version="$(curl -s https://www.jemarch.net/poke |
           pcregrep -o1 '>poke ([0-9.]+)</a>')"
-      update-source-version ${pname} "$new_version"
+      update-source-version poke "$new_version"
     '';
   };
 
   meta = with lib; {
```
Comment (AndersonTorres) [inline pkgs/applications/editors/poke/default.nix:82]: It deserves a removal (https://github.com/NixOS/nixpkgs/issues/208242)

```suggestion
  meta = {
```

File: pkgs/applications/editors/poke/default.nix [hunk 4]
```diff
@@ -97,18 +75,17 @@ in stdenv.mkDerivation rec {
...
     description = "Interactive, extensible editor for binary data";
     homepage = "http://www.jemarch.net/poke";
-    changelog = "https://git.savannah.gnu.org/cgit/poke.git/plain/ChangeLog?h=releases/poke-${version}";
+    changelog = "https://git.savannah.gnu.org/cgit/poke.git/plain/ChangeLog?h=releases/poke-${finalAttrs.version}";
     license = licenses.gpl3Plus;
     maintainers = with maintainers; [ AndersonTorres kira-bruneau ];
     platforms = platforms.unix;
```
Comment (AndersonTorres) [inline pkgs/applications/editors/poke/default.nix:88]:
```suggestion
    maintainers = with lib.maintainers; [ AndersonTorres kira-bruneau ];
    platforms = lib.platforms.unix;
```

**Review by AndersonTorres — CHANGES_REQUESTED** [review]

**Review by AndersonTorres — APPROVED** [review]

**Pull request merged**